### ISSUE TITLE
targetSlideNumber seems incorrect

### ISSUE THREAD

**Issue by kanduvisla** [iosscripts/iosslider#219]
When I fire a `onSlideStart`-function call, and read out the arguments. The `targetSlideNumber` seems incorrect.

I have a slider with 5 pictures and set my slider to autoSlide, so when I'm on currentSlide = 1, I expect the `targetSlideNumer` to be 2, from 2 to 3, 3 to 4, 4 to 5, 5 to 1, 1 to 2, etc...

But this is what happens (console.log output):

```
current: 1 target: 2 
current: 2 target: 4 
current: 3 target: 5 
current: 4 target: 5 
current: 5 target: 5 
current: 1 target: 3 
current: 2 target: 4 
current: 3 target: 5 
current: 4 target: 5 
current: 5 target: 5 
```

So something seems to go wrong somewhere, rendering the usability of `targetSlideNumber`... well... useless. Or am I missing something here?

edit: in addition, `prevSlideNumber` also seems to be not working (stays `0`)

**Comment by marcwhitbread** [comment]
Your right. It seems bugged. Not sure what is going wrong here. I will take a look at it and get back to you.

**Comment by marcwhitbread** [comment]
I'm going to pull this functionality until I can sort out how to handle it properly. All of the callbacks (start, change, complete, loaded) use the same variables, so when checking onSlideStart data, the data has already been updated by the onSlideChange/onSlideComplete functions. This renders the data useless.

It would probably only return usable data onSlideChange even if the data was being calculated correctly. I will most likely change the script to only return "target" data onSlideStart and "prev" data onSlideChange. But that just adds complexity for the end user.

**Comment by mems** [comment]
On version v1.2.19 (05/14/2013), a bug still there.

There a difference for `args.targetSlideNumber` between `slideChange` and `slideComplete`

```
slideStart
target: undefined prev: 1 current: 2
slideChange
target: 4 prev: 2 current: 3
slideComplete
target: 3 prev: 2 current: 3
```

**Comment by marcwhitbread** [comment]
@mems thanks for the submission. I can't seem to reproduce it.

Under what conditions did the error occur? Is this a manual slide or an automated slide (eg. goToSlide/autoslide function)? also, can I get your CSS for you slider elements (or the name of the demo if it's one of mine)?

**Comment by mems** [comment]
I use default demo with changing settings to:

```
$('.iosSlider').iosSlider({
    onSlideStart: function(args){console.log("slideStart\n", "target:", args.targetSlideNumber, "previous:", args.previousSlideNumber, "current:", args.currentSlideNumber)},
    onSlideChange: function(args){console.log("slideChange\n", "target:", args.targetSlideNumber, "previous:", args.previousSlideNumber, "current:", args.currentSlideNumber)},
    onSlideComplete: function(args){console.log("slideComplete\n", "target:", args.targetSlideNumber, "previous:", args.previousSlideNumber, "current:", args.currentSlideNumber)},
    autoSlideTimer: 1000,
    autoSlide: true,
    infiniteSlider: true
});
```

Launch it an wait 15 seconds (2 complete slides), watch logs. Tested with commit 38424f250dc89d8f197531d13779b80ee665bce3

If you remove `infiniteSlider` setting the problem disappears but for the 4th slide, `slideChange` event is trigger 3 times (for each, `currentSlideNumber` is equal respectively to 3, 2 and 1)

**Comment by marcwhitbread** [comment]
@mems Thanks for the info. I see it now. It seems to be exclusive to the autoslide function. I will try and get a fix out.

**Comment by marcwhitbread** [comment]
@mems I have applied a fix to the issue you reported in v1.2.24. Should be reporting the correct targetSlideNumber for onslidechange now.
